### ISSUE TITLE
Gradient Cumulation is friendly to FEW-GPU people

### ISSUE THREAD

**Issue by Harryoung** [open-mmlab/mmsegmentation#337]
# Describe the feature
As I know, gradient cumulation is an efficient way to achieve an equivalent big batchsize by gradient updating every few iterations,  which is very necessary for people who only have one or two GPUs, like me. o(╥﹏╥)o . The implementation seems not to be complex, so why not add this charming feature?
**Motivation** 
It is very useful for FEW-GPU ones!!! 

**Related resources**
This can be a reference: https://www.zhihu.com/question/375988016.

**Comment by Harryoung** [comment]
It seems that we just need to modify few lines of codes. See  
https://github.com/open-mmlab/mmcv/blob/276883f1205415fa97ea514f0198289740ddc4ea/mmcv/runner/hooks/optimizer.py#L13-L34
1. In the initiation, add one argument `cumul_step`, which is set 1 by default.
2. In the `after_train_iter`, add one condition: `if not runner._iter % self.cumul_step`, then put the original codes into the condition block.
3. Move the `runner.optimizer.zero_grad()` to the last line.
